accepted/2021/statics-in-interfaces/README.md

```diff
@@ -6,8 +6,1267 @@
+
+        // 5.7.2
+        //  enum Class(TSelf x);
+        //  bool IsSignMinus(TSelf x);
+        //  bool IsNormal(TSelf x);
+        //  bool IsFinite(TSelf x);
+        //  bool IsZero(TSelf x);
+        //  bool IsSubnormal(TSelf x);
+        //  bool IsInfinite(TSelf x);
+        //  bool IsNaN(TSelf x);
+        //  bool IsSignaling(TSelf x);
+        //  bool IsCanonical(TSelf x);
+        //  enum Radix(TSelf x);
+        //  bool TotalOrder(TSelf x, TSelf y);
+        //  bool TotalOrderMag(TSelf x, TSelf y);
+
+        // 9.4
+        //  TSelf Sum(TSelf[] x);
+        //  TSelf Dot(TSelf[] x, TSelf[] y);
+        //  TSelf SumSquare(TSelf[] x);
+        //  TSelf SumAbs(TSelf[] x);
+        //  (TSelf, TInt) ScaledProd(TSelf[] x);
+        //  (TSelf, TInt) ScaledProdSum(TSelf[] x, TSelf[] y);
+        //  (TSelf, TInt) ScaledProdDiff(TSelf[] x, TSelf[] y);
+
+        // 9.5
+        //  (TSelf, TSelf) AugmentedAddition(TSelf x, TSelf y);
+        //  (TSelf, TSelf) AugmentedSubtraction(TSelf x, TSelf y);
+        //  (TSelf, TSelf) AugmentedMultiplication(TSelf x, TSelf y);
+
+        // 9.6
+        //  TSelf Minimum(TSelf x, TSelf y);
+        //  TSelf MinimumNumber(TSelf x, TSelf y);
+        //  TSelf Maximum(TSelf x, TSelf y);
+        //  TSelf MaximumNumber(TSelf x, TSelf y);
+        //  TSelf MinimumMagnitude(TSelf x, TSelf y);
+        //  TSelf MinimumMagnitudeNumber(TSelf x, TSelf y);
+        //  TSelf MaximumMagnitude(TSelf x, TSelf y);
+        //  TSelf MaximumMagnitudeNumber(TSelf x, TSelf y);
+
+        // 9.7
+        //  TSelf GetPayload(TSelf x);
+        //  TSelf SetPayload(TSelf x);
+        //  TSelf SetPayloadSignaling(TSelf x);
+    }
+
+    public interface IBinaryFloatingPoint<TSelf>
+        : IBinaryNumber<TSelf>,
+          IFloatingPoint<TSelf>
+        where TSelf : IBinaryFloatingPoint<TSelf>
+    {
+    }
+
+    public interface IDecimalFloatingPoint<TSelf>
+        : IFloatingPoint<TSelf>
+        where TSelf : IDecimalFloatingPoint<TSelf>
+    {
+        // This interface is defined for convenience of viewing the IEEE requirements
+        // it would not actually be defined until the .NET libraries requires it
+
+        // The majority of the IEEE required operations are listed below
+        // This doesn't include the recommended operations such as sin, cos, acos, etc
+
+        // 5.3.2
+        //  TSelf Quantize(TSelf x, TSelf y);
+        //  TSelf Quantum(TSelf x);
+
+        // 5.5.2
+        //  TDecEnc EncodeDecimal(TSelf x);
+        //  TSelf DecodeDecimal(TDecEnc x);
+        //  TBinEnc EncodeBinary(TSelf x);
+        //  TSelf DecodeBinary(TBinEnc x);
+
+        // 5.7.3
+        //  bool SameQuantum(TSelf x, TSelf y);
+    }
+}
+```
+
+### Interface Implementors
+
+Various types in the System namespace would implement the above interfaces as follows.
+
+```csharp
+namespace System
+{
+    public struct Byte
+        : IBinaryInteger<byte>,
+          IConvertible,
```

Comment:
Might be worth denoting somehow that some of these interfaces are already implemented by these types.